sync command panic

<!--
Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->

**What happened**:
`old=redis://localhost/0 juicefs sync jfs://old/ /root/transferplace/raw/44234/ --threads 32`

```
root@ns380705:~# old=redis://localhost/0 juicefs sync jfs://old/ /root/transferplace/raw/44234/ --threads 32
2025/05/07 20:39:36.180054 juicefs[4062808] <INFO>: Meta address: redis://localhost/0 [NewClient@interface.go:578]
2025/05/07 20:39:36.181470 juicefs[4062808] <INFO>: Ping redis latency: 56.437µs [checkServerConfig@redis.go:3672]
2025/05/07 20:39:36.183152 juicefs[4062808] <INFO>: Prometheus metrics listening on 127.0.0.1:9567 [exposeMetrics@mount.go:134]
2025/05/07 20:39:36.183754 juicefs[4062808] <INFO>: Syncing from jfs://hath/ to file:///root/transferplace/raw/44234/ [Sync@sync.go:1699]
2025/05/07 20:39:36.503382 juicefs[4062808] <WARNING>: fail to read sliceId 8655350 (off:0, size:760461, clen: 760461, inode: 2161117): get chunks/8/8655/8655350_0_760461: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655350_0_760461: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:36.509137 juicefs[4062808] <WARNING>: fail to read sliceId 8655350 (off:0, size:760461, clen: 760461, inode: 2161117): get chunks/8/8655/8655350_0_760461: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655350_0_760461: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:36.799524 juicefs[4062808] <WARNING>: fail to read sliceId 8655336 (off:0, size:195404, clen: 195404, inode: 2161116): get chunks/8/8655/8655336_0_195404: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655336_0_195404: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:36.815813 juicefs[4062808] <WARNING>: fail to read sliceId 8655350 (off:0, size:760461, clen: 760461, inode: 2161117): get chunks/8/8655/8655350_0_760461: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655350_0_760461: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:36.815923 juicefs[4062808] <WARNING>: fail to read sliceId 8655336 (off:0, size:195404, clen: 195404, inode: 2161116): get chunks/8/8655/8655336_0_195404: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655336_0_195404: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.118801 juicefs[4062808] <WARNING>: fail to read sliceId 8655336 (off:0, size:195404, clen: 195404, inode: 2161116): get chunks/8/8655/8655336_0_195404: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655336_0_195404: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.417209 juicefs[4062808] <WARNING>: fail to read sliceId 8655350 (off:0, size:760461, clen: 760461, inode: 2161117): get chunks/8/8655/8655350_0_760461: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655350_0_760461: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.474081 juicefs[4062808] <WARNING>: fail to read sliceId 8655179 (off:0, size:36352, clen: 36352, inode: 2161110): get chunks/8/8655/8655179_0_36352: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655179_0_36352: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.474322 juicefs[4062808] <WARNING>: fail to read sliceId 8655177 (off:0, size:311382, clen: 311382, inode: 2161110): get chunks/8/8655/8655177_0_311382: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655177_0_311382: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.474353 juicefs[4062808] <WARNING>: fail to read sliceId 8655178 (off:0, size:124152, clen: 124152, inode: 2161110): get chunks/8/8655/8655178_0_124152: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655178_0_124152: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.492435 juicefs[4062808] <WARNING>: fail to read sliceId 8655179 (off:0, size:36352, clen: 36352, inode: 2161110): get chunks/8/8655/8655179_0_36352: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655179_0_36352: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.495996 juicefs[4062808] <WARNING>: fail to read sliceId 8655177 (off:0, size:311382, clen: 311382, inode: 2161110): get chunks/8/8655/8655177_0_311382: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655177_0_311382: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.496142 juicefs[4062808] <WARNING>: fail to read sliceId 8655178 (off:0, size:124152, clen: 124152, inode: 2161110): get chunks/8/8655/8655178_0_124152: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655178_0_124152: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.735187 juicefs[4062808] <WARNING>: fail to read sliceId 8655336 (off:0, size:195404, clen: 195404, inode: 2161116): get chunks/8/8655/8655336_0_195404: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655336_0_195404: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.762044 juicefs[4062808] <WARNING>: fail to read sliceId 22520023 (off:0, size:211118, clen: 211118, inode: 5174982): get chunks/22/22520/22520023_0_211118: open /root/transferplace/hath/44234/hath/chunks/22/22520/22520023_0_211118: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.763285 juicefs[4062808] <WARNING>: fail to read sliceId 22520023 (off:0, size:211118, clen: 211118, inode: 5174982): get chunks/22/22520/22520023_0_211118: open /root/transferplace/hath/44234/hath/chunks/22/22520/22520023_0_211118: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.798835 juicefs[4062808] <WARNING>: fail to read sliceId 8655177 (off:0, size:311382, clen: 311382, inode: 2161110): get chunks/8/8655/8655177_0_311382: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655177_0_311382: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.798899 juicefs[4062808] <WARNING>: fail to read sliceId 8655179 (off:0, size:36352, clen: 36352, inode: 2161110): get chunks/8/8655/8655179_0_36352: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655179_0_36352: no such file or directory [readSlice@reader.go:826]
2025/05/07 20:39:37.798926 juicefs[4062808] <WARNING>: fail to read sliceId 8655178 (off:0, size:124152, clen: 124152, inode: 2161110): get chunks/8/8655/8655178_0_124152: open /root/transferplace/hath/44234/hath/chunks/8/8655/8655178_0_124152: no such file or directory [readSlice@reader.go:826]
panic: runtime error: index out of range [0] with length 0

goroutine 4507 [running]:
github.com/juicedata/juicefs/pkg/object.(*dataEncryptor).Decrypt(0x4a2f4e0?, {0xc0020f6000?, 0x1?, 0x0?})
        /go/src/github.com/juicedata/juicefs/pkg/object/encrypt.go:197 +0x369
github.com/juicedata/juicefs/pkg/object.(*encrypted).Get(0xc0012338c0, {0xc000916390?, 0xc002968340?}, 0x0, 0xffffffffffffffff, {0xc0020e7b90?, 0x4000000000000?, 0x2a82a5aa6ea569aa?})
        /go/src/github.com/juicedata/juicefs/pkg/object/encrypt.go:242 +0x122
github.com/juicedata/juicefs/pkg/chunk.(*cachedStore).load.func2()
        /go/src/github.com/juicedata/juicefs/pkg/chunk/cached_store.go:741 +0x1cc
github.com/juicedata/juicefs/pkg/utils.WithTimeout.func1()
        /go/src/github.com/juicedata/juicefs/pkg/utils/utils.go:102 +0x2c
created by github.com/juicedata/juicefs/pkg/utils.WithTimeout in goroutine 4504
        /go/src/github.com/juicedata/juicefs/pkg/utils/utils.go:101 +0xc9
```


**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**
```
root@ns380705:~# juicefs status redis://localhost/0
2025/05/07 20:40:30.594350 juicefs[4066897] <INFO>: Meta address: redis://localhost/0 [NewClient@interface.go:578]
2025/05/07 20:40:30.595896 juicefs[4066897] <INFO>: Ping redis latency: 112.516µs [checkServerConfig@redis.go:3672]
{
  "Setting": {
    "Name": "hath",
    "UUID": "72d4ddbf-f0dc-49a8-898a-a0e80241aec2",
    "Storage": "file",
    "Bucket": "/root/transferplace/hath/44234/",
    "BlockSize": 4096,
    "Compression": "none",
    "EncryptKey": "removed",
    "KeyEncrypted": true,
    "TrashDays": 1,
    "MetaVersion": 1,
    "EnableACL": false
  },
  "Sessions": [],
  "Stat": {
    "UsedSpace": 1338870272000,
    "AvailableSpace": 1124561036570624,
    "UsedInodes": 3264334,
    "AvailableInodes": 10485760
  }
}
```


**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:juicefs version 1.3.0-beta1+2025-04-22.0ae8ccd ,also tried v1.2.3, still same

- Cloud provider or hardware configuration running JuiceFS: ovh dedicated server
- OS (e.g `cat /etc/os-release`):Ubuntu 22.04.5 LTS
- Kernel (e.g. `uname -a`):Linux ns380705 5.15.0-134-generic # 145-Ubuntu SMP Wed Feb 12 20:08:39 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux
- Object storage (cloud provider and region, or self maintained):all local,also tried sync to localhost s3,same error
- Metadata engine info (version, cloud provider managed or self maintained):dragonfly
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):all local
- Others:

--- a/pkg/object/encrypt.go
+++ b/pkg/object/encrypt.go
@@ -194,6 +194,9 @@ func (e *dataEncryptor) Encrypt(plaintext []byte) ([]byte, error) {
 }
 
 func (e *dataEncryptor) Decrypt(ciphertext []byte) ([]byte, error) {
+	if len(ciphertext) < 3 {
+		return nil, fmt.Errorf("received encrypted text length is less than 3, the object is corrupted")
+	}
 	keyLen := int(ciphertext[0])<<8 + int(ciphertext[1])
 	nonceLen := int(ciphertext[2])
 	if 3+keyLen+nonceLen >= len(ciphertext) {

--- a/pkg/object/encrypt_test.go
+++ b/pkg/object/encrypt_test.go
@@ -25,6 +25,7 @@ import (
 	"io"
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 )
 
@@ -222,6 +223,11 @@ func TestEncryptedStore(t *testing.T) {
 	r, _ = es.Get("a", 0, -1)
 	d, _ = io.ReadAll(r)
 	if string(d) != "hello" {
+		t.Fail()
+	}
+	_ = s.Put("emptyfile", bytes.NewReader([]byte("")))
+	_, err = es.Get("emptyfile", 0, -1)
+	if err == nil || !strings.Contains(err.Error(), "the object is corrupted") {
 		t.Fail()
 	}
 }

